### PR TITLE
Support to update local user details if the caller is either the same user or admin

### PR DESCRIPTION
**Pull request by yuva29**
Re-opening PR https://github.com/contiv/auth_proxy/pull/89 again  as i'm not able update the old one due to visibility changes made to the repo. 

Signed-off-by: Yuva Shankar yuva29@users.noreply.github.com

### CODE REVIEW

**Comment by yuva29** [comment]
@rhim @dseevr Sorry for the inconvenience. PTAL 😄 

Also, I addressed the potential issue that may arise if the same username exists in local and ldap system.

File: systemtests/local_user_test.go [hunk 1]
```diff
@@ -103,6 +103,17 @@ func (s *systemtestSuite) userUpdate(c *C) {
 
 			// try login again using new password
 			_ = loginAs(c, username, "test")
+
+			// userUpdateEndpoint is not adminOnly any more. It can be accessed by user himself.
```
Comment (rhim) [inline systemtests/local_user_test.go]: Nit: this comment is not very helpful. Just make is say that it is admin or calling user. Recording history in comment is usually good only for a short time.

File: systemtests/rbac_test.go [hunk 2]
```diff
@@ -480,29 +480,67 @@ func (s *systemtestSuite) TestAdminRoleRequired(c *C) {
 		// login as username, should succeed
 		testuserToken := loginAs(c, username, username)
 
-		// try calling an admin api (e.g., update user) using test user token
-		// This should fail with forbidden since user doesn't have admin access
+		// update user is no longer adminOnly. so update using the user's token should pass
```
Comment (rhim) [inline systemtests/rbac_test.go]: same as above.

File: systemtests/rbac_test.go [hunk 3]
```diff
@@ -480,29 +480,67 @@ func (s *systemtestSuite) TestAdminRoleRequired(c *C) {
...
+		// Only ADD and DELETE is adminOnly now.
```
Comment (rhim) [inline systemtests/rbac_test.go]: same as above.

File: systemtests/rbac_test.go [hunk 4]
```diff
@@ -480,29 +480,67 @@ func (s *systemtestSuite) TestAdminRoleRequired(c *C) {
...
 		data := `{"first_name":"Temp", "last_name": "User"}`
 		endpoint := proxy.V1Prefix + "/local_users/" + username
 		resp, _ := proxyPatch(c, testuserToken, endpoint, []byte(data))
+		// StatusCode is no longer 403 as the user is allowed to update his/her details
```
Comment (rhim) [inline systemtests/rbac_test.go]: same as above.

File: systemtests/rbac_test.go [hunk 5]
```diff
@@ -480,29 +480,67 @@ func (s *systemtestSuite) TestAdminRoleRequired(c *C) {
...
 		data = `{"first_name":"Temp", "last_name": "User"}`
 		respBody := `{"username":"` + username + `","first_name":"Temp","last_name":"User","disable":false}`
 		s.updateLocalUser(c, username, data, respBody, testuserToken)
 
 		// delete authorization
 		s.deleteAuthorization(c, authz.AuthzUUID, adToken)
 
-		// calling admin api should fail again withouth requiring new token (since cached value
-		// of role authz in token isn't used)
+		// update user is no longer adminOnly. so update using the user token should pass
```
Comment (rhim) [inline systemtests/rbac_test.go]: same as above.

**Pull request merged**